test/setup.js

```diff
@@ -0,0 +1,84 @@
+import head from 'lodash/head';
+import last from 'lodash/last';
+import mergeWith from 'lodash/mergeWith';
+import reduce from 'lodash/reduce';
+import tail from 'lodash/tail';
+
+const files = [];
+let whitelistConfig;
+
+function enableWhitelist() {
+	whitelistConfig = require( 'tests.json' );
+}
+
+function addFile( file ) {
+	if ( whitelistConfig ) {
+		const pathParts = tail( file.split( '/' ) );
+
+		if ( ! isFileWhitelisted( whitelistConfig, pathParts ) ) {
+			console.log( `Skipping file: ${file}.` );
+			return;
+		}
+	}
+	files.push( file );
+}
+
+function isFileWhitelisted( config, pathParts ) {
+	const folder = head( pathParts );
+
+	if ( config[ folder ] ) {
+		if ( folder === 'test' ) {
+			return ( config[ folder ].indexOf( getFileName( pathParts ) ) !== false );
+		}
+
+		return isFileWhitelisted( config[ folder ], tail( pathParts ) );
+	}
+
+	return false;
+}
+
+function getConfig() {
+	if ( ! whitelistConfig && files.length === 0 ) {
+		// this assumes that there's a tests.json at the root of NODE_PATH
+		console.log( 'No valid tests provided, loading whitelisted tests.' );
+		return require( 'tests.json' );
+	}
+
+	return filesToConfig();
+}
+
+function filesToConfig() {
+	return reduce( files, ( config, file ) => {
+		const fileConfig = fileToConfig( tail( file.split( '/' ) ) );
```

Comment:
could probably do this with lastIndexOf and avoid creating the large array just to take the last item